scene/main/viewport.cpp

```diff
@@ -2355,6 +2341,20 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 	}
 
 	if (mm.is_null() && mb.is_null() && p_event->is_action_type()) {
+		const bool resize_allowed = !DisplayServer::get_singleton()->window_get_flag(DisplayServer::WINDOW_FLAG_RESIZE_DISABLED);
+
+		if (p_event->is_pressed() && p_event->is_action_pressed("ui_toggle_fullscreen") && resize_allowed) {
```

Comment:
```suggestion
		if (resize_allowed && p_event->is_pressed() && p_event->is_action_pressed("ui_toggle_fullscreen")) {
```
This way we can skip parsing input if resize is disabled.